specs/data_structures.md

```diff
@@ -320,31 +286,178 @@ A proof into an SMT is structured as:
+
+These shares are arranged in the [first quadrant](#2d-reed-solomon-encoding-scheme) (`Q0`) of the `AVAILABLE_DATA_MATRIX_HALFSIZE*2 * AVAILABLE_DATA_MATRIX_HALFSIZE*2` available data matrix in _row-major_ order. In the example below, each reserved data element takes up exactly one share.
+
+![fig: Original data: reserved.](figures/rs2d_originaldata_reserved.svg)
+
+Each message in the list `messageData` is independently serialized and split into `SHARE_SIZE` shares. Then each message is padded by appending zero-shares (i.e. `SHARE_SIZE` bytes of `0x00`) until the smallest enclosing power of 2 (e.g. a message that fits into three shares is padded to four shares). For each message, the following algorithm is used to place it in the available data matrix, with row-major order:
+1. Place the first share of the message at the next unused location in the matrix whose column in aligned with the number of padded shares (i.e. if there are four padded shares, then only every fourth location can be used to start) **unless** there are insufficient unused locations in the row. Then place the remaining shares in the following locations.
+1. If there are insufficient unused locations in the row, place the first share of the message at the first column of the next row. Then place the remaining shares in the following locations.
+
+In the example below, two messages (of padded sizes 2 and 1) are placed following the aforementioned rules.
+
+![fig: Original data: messages.](figures/rs2d_originaldata_message.svg)
+
+By construction, this gives a useful property: transactions [can commit to a commitment to a list of roots](#transaction) that are each guaranteed (assuming the block is valid) to be subtree roots in one or more of the row NMTs.
+
+# Available Data
 
 ## TransactionData
 
-Wrapper for transaction data, which is a simple list of [Transaction](#transaction)s.
+| name                  | type                                          | description                   |
+| --------------------- | --------------------------------------------- | ----------------------------- |
+| `wrappedTransactions` | [WrappedTransaction](#wrappedtransaction)`[]` | List of wrapped transactions. |
 
- | name           | type                            | description           |
- | -------------- | ------------------------------- | --------------------- |
- | `transactions` | [Transaction](#transaction)`[]` | List of transactions. |
+### WrappedTransaction
```

Comment:
I think it would be good to motivate the `wrapped` types in a sentence before.
Sth like: In order to be able to lookup the corresponding messages a Tx pays for in the "data matrix" we introduce the notion of wrapped Tx.